### PR TITLE
feat(mgmt): exponential polling interval

### PR DESCRIPTION
**Pull request by archerzz**
1. refactor to implement various polling strategy for `OperationInternalBase`
2. add property `OperationOrResponseInternals.PollingStrategy` to override default polling strategy
3. update `MgmtLongRunningOperationWriter` to set exponential polling strategy for mgmt plane

resolve #26922

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/assets/Azure.Core.Shared/OperationInternalBase.cs [hunk 1]
```diff
@@ -58,6 +59,22 @@ protected OperationInternalBase(ClientDiagnostics clientDiagnostics, Response ra
         /// </summary>
         public TimeSpan DefaultPollingInterval { get; set; }
```
Comment (archerzz) [inline src/assets/Azure.Core.Shared/OperationInternalBase.cs:60]: I need more time to figure out if we can remove `DefaultPollingInterval`. Although it's public, it seems not be explicitly set (e.g. no place really set a non-default value) and the getter is invoked in polling implementation.
If we could remove it, then the logic of `PollingInterval` (see below) can be simplified.

File: test/TestProjects/ExactMatchFlattenInheritance/Generated/LongRunningOperation/ExactMatchFlattenInheritanceArmOperation.cs [hunk 2]
```diff
@@ -29,11 +29,13 @@ protected ExactMatchFlattenInheritanceArmOperation()
         internal ExactMatchFlattenInheritanceArmOperation(Response response)
         {
             _operation = new OperationOrResponseInternals(response);
+            _operation.PollingStrategy = new ExponentialPollingStrategy();
```
Comment (archerzz) [inline test/TestProjects/ExactMatchFlattenInheritance/Generated/LongRunningOperation/ExactMatchFlattenInheritanceArmOperation.cs:32]: Another options is that we add constructors of `OperationOrResponseInternals` for mgmt plane. And invoke the specific constructor here.

File: src/assets/Azure.Core.Shared/OperationInternalBase.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/assets/Azure.Core.Shared/OperationInternalBase.cs:60]: This type is defined in Azure.Core, so the change should happen there.

**Comment by archerzz** [comment]
This PR has been covered in https://github.com/Azure/autorest.csharp/pull/2000

**Pull request closed without merging**